lib/archethic/db/embedded_impl/chain_writer.ex

```diff
@@ -45,9 +45,29 @@ defmodule Archethic.DB.EmbeddedImpl.ChainWriter do
 
   def handle_call(
         {:append_tx, genesis_address, tx},
-        _from,
+        from,
         state = %{db_path: db_path}
       ) do
+    %Task{ref: ref} = Task.async(fn -> write_transaction(genesis_address, tx, db_path) end)
+    {:noreply, Map.update!(state, :clients, &Map.put(&1, ref, from))}
+  end
+
+  def handle_info({ref, :ok}, state) do
+    case pop_in(state, [:clients, ref]) do
+      {nil, ^state} ->
+        {:noreply, state}
+
+      {from, new_state} ->
+        GenServer.reply(from, :ok)
```

Comment:
I think, we should add  test cases for it.

Reply:
If the existing test cases are working, it mean we didn't break something. 
We could add more load testing